# Carregamento da Base Tratada

Nesta etapa é carregado o dataset após o processo de limpeza e classificação realizado na etapa de preparação dos dados.

A base contém documentos técnicos extraídos de repositórios, classificados por categoria e preparados para geração dos embeddings utilizados pelo modelo de IA.

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [46]:
import pandas as pd

caminho_dataset = "/content/drive/MyDrive/Hackathon_ONE/dataset/tratado/documentation_tratado.parquet"

dataset = pd.read_parquet(caminho_dataset)

print("Dataset carregado!")
print(dataset.shape)

dataset.head()

Dataset carregado!
(2879, 10)


,id,repo_full_name,file_name,content,file_size,readme_quality_score,collected_at,num_caracteres,conteudo_repetido,categoria_documento
0,2454,0xN1nja/homepagectl,README.md,# homepagectl\n\nTired of manually writing hom...,2041,42,2026-06-27 19:14:23.602441,2041,False,Documentacao
1,2455,0xN1nja/homepagectl,go.mod,module github.com/0xN1nja/homepagectl\n\ngo 1....,242,0,2026-06-27 19:14:43.368223,242,False,Configuracao_Linguagem
2,2158,0xNyk/council-of-high-intelligence,CHANGELOG.md,# Changelog\n\nAll notable changes to this pro...,5454,0,2026-06-27 18:22:28.437941,5454,False,Historico_Alteracoes
3,2157,0xNyk/council-of-high-intelligence,CONTRIBUTING.md,# Contributing\n\nThanks for contributing to C...,941,0,2026-06-27 18:22:26.587716,941,False,Contribuicao
4,2156,0xNyk/council-of-high-intelligence,LICENSE,MIT License\n\nCopyright (c) 2026 nyk\n\nPermi...,1060,0,2026-06-27 18:22:24.846284,1060,False,Licenca


In [3]:
import pandas as pd

caminho = "/content/drive/MyDrive/Hackathon_ONE/dataset/tratado/documentation_tratado.parquet"

df = pd.read_parquet(caminho)

print(df.shape)
df.head()

(2879, 10)


,id,repo_full_name,file_name,content,file_size,readme_quality_score,collected_at,num_caracteres,conteudo_repetido,categoria_documento
0,2454,0xN1nja/homepagectl,README.md,# homepagectl\n\nTired of manually writing hom...,2041,42,2026-06-27 19:14:23.602441,2041,False,Documentacao
1,2455,0xN1nja/homepagectl,go.mod,module github.com/0xN1nja/homepagectl\n\ngo 1....,242,0,2026-06-27 19:14:43.368223,242,False,Configuracao_Linguagem
2,2158,0xNyk/council-of-high-intelligence,CHANGELOG.md,# Changelog\n\nAll notable changes to this pro...,5454,0,2026-06-27 18:22:28.437941,5454,False,Historico_Alteracoes
3,2157,0xNyk/council-of-high-intelligence,CONTRIBUTING.md,# Contributing\n\nThanks for contributing to C...,941,0,2026-06-27 18:22:26.587716,941,False,Contribuicao
4,2156,0xNyk/council-of-high-intelligence,LICENSE,MIT License\n\nCopyright (c) 2026 nyk\n\nPermi...,1060,0,2026-06-27 18:22:24.846284,1060,False,Licenca


In [4]:
import sentence_transformers

print("Biblioteca encontrada!")

Biblioteca encontrada!


In [5]:
df_teste = df.head(100).copy()

print(df_teste.shape)

(100, 10)


In [6]:
from sentence_transformers import SentenceTransformer

modelo = SentenceTransformer("all-MiniLM-L6-v2")

print("Modelo carregado com sucesso!")

modules.json:   0%|          | 0.00/349 [00:00<?, ?B/s]

config_sentence_transformers.json:   0%|          | 0.00/116 [00:00<?, ?B/s]

README.md:   0%|          | 0.00/10.5k [00:00<?, ?B/s]

sentence_bert_config.json:   0%|          | 0.00/53.0 [00:00<?, ?B/s]

config.json:   0%|          | 0.00/612 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/90.9M [00:00<?, ?B/s]

Loading weights:   0%|          | 0/103 [00:00<?, ?it/s]

tokenizer_config.json:   0%|          | 0.00/350 [00:00<?, ?B/s]

vocab.txt:   0%|          | 0.00/232k [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/466k [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/112 [00:00<?, ?B/s]

config.json:   0%|          | 0.00/190 [00:00<?, ?B/s]

Modelo carregado com sucesso!


In [7]:
df_readme = df[df["categoria_documento"] == "Documentacao"].copy()

print(df_readme.shape)

(941, 10)


In [8]:
# Seleciona apenas 10 documentos para teste
df_teste = df_readme.head(10).copy()

# Gera os embeddings
embeddings = modelo.encode(
    df_teste["content"].tolist(),
    show_progress_bar=True
)

print("Embeddings gerados com sucesso!")
print("Quantidade de embeddings:", len(embeddings))
print("Dimensão de cada embedding:", len(embeddings[0]))

Batches:   0%|          | 0/1 [00:00<?, ?it/s]

Embeddings gerados com sucesso!
Quantidade de embeddings: 10
Dimensão de cada embedding: 384


In [9]:
from sklearn.metrics.pairwise import cosine_similarity

print("Biblioteca carregada!")

Biblioteca carregada!


In [10]:
similaridade = cosine_similarity(embeddings)

print(similaridade.shape)

similaridade

(10, 10)


array([[1.        , 0.1200543 , 0.07853928, 0.0571155 , 0.01085536,
        0.08669773, 0.07934102, 0.14201044, 0.13217494, 0.18139133],
       [0.1200543 , 1.0000002 , 0.22885185, 0.45449188, 0.20833676,
        0.1301941 , 0.18063387, 0.39304507, 0.21394749, 0.54344106],
       [0.07853928, 0.22885185, 1.        , 0.29520547, 0.2195858 ,
        0.22365475, 0.17622365, 0.29086882, 0.13204233, 0.14632614],
       [0.0571155 , 0.45449188, 0.29520547, 1.        , 0.1338211 ,
        0.1827314 , 0.12972978, 0.31408387, 0.190594  , 0.30723602],
       [0.01085536, 0.20833676, 0.2195858 , 0.1338211 , 1.0000001 ,
        0.18286148, 0.11439822, 0.17881796, 0.01284742, 0.2988793 ],
       [0.08669773, 0.1301941 , 0.22365475, 0.1827314 , 0.18286148,
        1.        , 0.123655  , 0.1620292 , 0.14049774, 0.18308823],
       [0.07934102, 0.18063387, 0.17622365, 0.12972978, 0.11439822,
        0.123655  , 0.99999994, 0.2149414 , 0.10954282, 0.26623124],
       [0.14201044, 0.39304507, 0.2908688

In [13]:
import numpy as np

def recomendar_documentos(indice, top_n=3):
    similaridades = similaridade[indice]

    # Ordena do maior para o menor
    indices = np.argsort(similaridades)[::-1]

    # Remove o próprio documento
    indices = indices[indices != indice]

    print(f"\nDocumento consultado:\n")
    print(df_teste.iloc[indice]["repo_full_name"])
    print(df_teste.iloc[indice]["file_name"])

    print("\nDocumentos mais semelhantes:\n")

    for i in indices[:top_n]:
        print(f"Repositório: {df_teste.iloc[i]['repo_full_name']}")
        print(f"Arquivo: {df_teste.iloc[i]['file_name']}")
        print(f"Similaridade: {similaridades[i]:.3f}")
        print("-" * 50)

In [14]:
recomendar_documentos(0)


Documento consultado:

0xN1nja/homepagectl
README.md

Documentos mais semelhantes:

Repositório: Aider-AI/aider
Arquivo: README.md
Similaridade: 0.181
--------------------------------------------------
Repositório: ARMmbed/mbed-os
Arquivo: README.md
Similaridade: 0.142
--------------------------------------------------
Repositório: Ahmed-Ali/JSONExport
Arquivo: README.md
Similaridade: 0.132
--------------------------------------------------


# Geração dos Embeddings - SCRIPTO

In [17]:
print(df.shape)
print(df_readme.shape)
print(type(modelo))

(2879, 10)
(941, 10)
<class 'sentence_transformers.sentence_transformer.model.SentenceTransformer'>


In [18]:
# Gerando embeddings de todos os documentos técnicos

conteudos = df_readme["content"].tolist()

embeddings_readme = modelo.encode(
    conteudos,
    show_progress_bar=True,
    batch_size=32
)

print("Embeddings gerados com sucesso!")
print("Quantidade de documentos:", len(embeddings_readme))
print("Dimensão dos vetores:", len(embeddings_readme[0]))

Batches:   0%|          | 0/30 [00:00<?, ?it/s]

Embeddings gerados com sucesso!
Quantidade de documentos: 941
Dimensão dos vetores: 384


In [19]:
import os

caminho_base = "/content/drive/MyDrive/Hackathon_ONE/SCRIPTO_AI"

os.makedirs(f"{caminho_base}/embeddings", exist_ok=True)
os.makedirs(f"{caminho_base}/metadados", exist_ok=True)

print("Estrutura SCRIPTO_AI criada com sucesso!")

Estrutura SCRIPTO_AI criada com sucesso!


In [20]:
import numpy as np

caminho_embeddings = "/content/drive/MyDrive/Hackathon_ONE/SCRIPTO_AI/embeddings/embeddings_readme.npy"

np.save(caminho_embeddings, embeddings_readme)

print("Embeddings salvos com sucesso!")

Embeddings salvos com sucesso!


In [21]:
metadados = df_readme[
    [
        "repo_full_name",
        "file_name",
        "categoria_documento",
        "content"
    ]
].copy()

metadados["indice_embedding"] = range(len(metadados))

caminho_metadados = "/content/drive/MyDrive/Hackathon_ONE/SCRIPTO_AI/metadados/documentos_readme.csv"

metadados.to_csv(caminho_metadados, index=False)

print("Metadados salvos com sucesso!")
print(metadados.shape)

Metadados salvos com sucesso!
(941, 5)


In [22]:
import os

caminhos = [
    "/content/drive/MyDrive/Hackathon_ONE/SCRIPTO_AI/embeddings/embeddings_readme.npy",
    "/content/drive/MyDrive/Hackathon_ONE/SCRIPTO_AI/metadados/documentos_readme.csv"
]

for arquivo in caminhos:
    print(arquivo)
    print("Existe:", os.path.exists(arquivo))
    print("-" * 40)

/content/drive/MyDrive/Hackathon_ONE/SCRIPTO_AI/embeddings/embeddings_readme.npy
Existe: True
----------------------------------------
/content/drive/MyDrive/Hackathon_ONE/SCRIPTO_AI/metadados/documentos_readme.csv
Existe: True
----------------------------------------


# Motor de Busca Semântica - SCRIPTO

Nesta etapa será criado o motor de busca inteligente do SCRIPTO.

Após a geração dos embeddings dos documentos técnicos, será realizada a preparação dos arquivos necessários para consulta.

O objetivo desta etapa é permitir que uma busca em linguagem natural seja comparada semanticamente com os documentos existentes, identificando conteúdos relacionados mesmo quando não possuem exatamente as mesmas palavras-chave.

Fluxo da etapa:

1. Carregar os embeddings gerados pelo modelo de IA;
2. Carregar os metadados dos documentos;
3. Associar cada vetor ao seu respectivo documento;
4. Preparar a base para cálculo de similaridade.

Os embeddings representam o conteúdo textual dos documentos em formato numérico, permitindo que o sistema encontre relações de significado entre diferentes conteúdos técnicos.

In [23]:
from sklearn.metrics.pairwise import cosine_similarity
import numpy as np
import pandas as pd

# Carregar embeddings salvos
caminho_embeddings = "/content/drive/MyDrive/Hackathon_ONE/SCRIPTO_AI/embeddings/embeddings_readme.npy"

embeddings_base = np.load(caminho_embeddings)

# Carregar metadados
caminho_metadados = "/content/drive/MyDrive/Hackathon_ONE/SCRIPTO_AI/metadados/documentos_readme.csv"

metadados = pd.read_csv(caminho_metadados)

print("Base de IA carregada!")
print("Embeddings:", embeddings_base.shape)
print("Metadados:", metadados.shape)

Base de IA carregada!
Embeddings: (941, 384)
Metadados: (941, 5)


# Função de Busca Semântica - SCRIPTO

Nesta etapa será desenvolvida a função responsável pela consulta inteligente dos documentos técnicos.

O usuário poderá realizar uma pesquisa utilizando linguagem natural e o modelo de IA irá comparar o significado da busca com os embeddings dos documentos existentes.

Funcionamento:

1. O texto informado pelo usuário será transformado em um embedding utilizando o mesmo modelo aplicado nos documentos;
2. Será calculada a similaridade entre a busca e todos os documentos armazenados;
3. Os documentos com maior proximidade semântica serão retornados como recomendação.

Essa abordagem permite encontrar conteúdos relacionados mesmo quando as palavras utilizadas na busca são diferentes das palavras presentes nos documentos.

Exemplo:

Busca:
"Como trabalhar com containers"

Pode encontrar documentos relacionados a:
"Docker", "containerização", "deploy de aplicações", entre outros.

In [25]:
def buscar_scripto(pergunta, top_n=5):

    # Transformar a pergunta em embedding
    embedding_pergunta = modelo.encode([pergunta])

    # Calcular similaridade com todos os documentos
    similaridades = cosine_similarity(
        embedding_pergunta,
        embeddings_base
    )[0]

    # Pegar os maiores valores
    indices = np.argsort(similaridades)[::-1][:top_n]

    resultados = metadados.iloc[indices].copy()

    resultados["similaridade"] = similaridades[indices]

    return resultados[
        [
            "repo_full_name",
            "file_name",
            "categoria_documento",
            "similaridade"
        ]
    ]

# Teste do Motor de Busca Semântica - SCRIPTO

Nesta etapa será realizada a validação do modelo de IA através de consultas em linguagem natural.

O objetivo é verificar se o sistema consegue recuperar documentos tecnicamente relacionados ao termo pesquisado utilizando similaridade semântica.

Diferente de uma busca tradicional por palavras-chave, o modelo considera o contexto e o significado dos textos.

Exemplo:

Consulta:
"docker containers"

O sistema deve retornar documentos relacionados a containers, infraestrutura, configuração ou tecnologias próximas.

In [26]:
buscar_scripto("docker containers")

,repo_full_name,file_name,categoria_documento,similaridade
351,crccheck/docker-hello-world,README.md,Documentacao,0.547565
399,eBayClassifiedsGroup/PanteraS,README.md,Documentacao,0.518441
830,stilleshan/frpc,README.md,Documentacao,0.511281
842,svhawks/Sendy-Docker,README.md,Documentacao,0.510385
627,mediacms-io/mediacms,README.md,Documentacao,0.461526


# Padronização da Resposta para Integração com API

Nesta etapa será ajustado o formato de retorno da busca semântica.

O objetivo é disponibilizar uma estrutura mais simples para consumo pelo Backend do SCRIPTO.

A resposta será organizada contendo:

- Nome do projeto;
- Tipo do documento encontrado;
- Categoria do conteúdo;
- Score de similaridade.

Esse formato facilita a comunicação entre o modelo de IA desenvolvido em Python e a API desenvolvida em Spring Boot.

In [27]:
def buscar_scripto(pergunta, top_n=5):

    embedding_pergunta = modelo.encode([pergunta])

    similaridades = cosine_similarity(
        embedding_pergunta,
        embeddings_base
    )[0]

    indices = np.argsort(similaridades)[::-1][:top_n]

    resultados = []

    for indice in indices:
        resultados.append({
            "projeto": metadados.iloc[indice]["repo_full_name"],
            "documento": metadados.iloc[indice]["file_name"],
            "categoria": metadados.iloc[indice]["categoria_documento"],
            "similaridade": round(float(similaridades[indice]) * 100, 2)
        })

    return resultados

In [28]:
buscar_scripto("docker containers")

[{'projeto': 'crccheck/docker-hello-world',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 54.76},
 {'projeto': 'eBayClassifiedsGroup/PanteraS',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 51.84},
 {'projeto': 'stilleshan/frpc',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 51.13},
 {'projeto': 'svhawks/Sendy-Docker',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 51.04},
 {'projeto': 'mediacms-io/mediacms',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 46.15}]

# Conclusão - Modelo IA SCRIPTO

Nesta etapa foi desenvolvido o mecanismo de busca semântica do SCRIPTO.

O modelo de IA foi capaz de transformar documentos técnicos em representações vetoriais e realizar consultas baseadas em significado.

Principais entregas:

- 941 documentos processados;
- Embeddings gerados utilizando Sentence Transformers;
- Busca semântica implementada;
- Ranking por similaridade;
- Estrutura de resposta preparada para integração com API.

Os arquivos gerados podem ser utilizados pelo Backend para disponibilizar a funcionalidade de pesquisa inteligente no sistema SCRIPTO.

# Sugestão Automática de Conteúdos Relacionados - SCRIPTO

Nesta etapa será implementada uma funcionalidade adicional utilizando os embeddings já gerados.

O objetivo é recomendar documentos técnicos semelhantes ao conteúdo atualmente consultado pelo usuário.

Diferente da busca tradicional, essa recomendação utiliza similaridade semântica, analisando o contexto dos documentos para identificar relações entre conteúdos.

Fluxo:

1. Usuário seleciona um documento;
2. O sistema utiliza o embedding correspondente;
3. Calcula a similaridade com os demais documentos da base;
4. Remove o próprio documento da recomendação;
5. Retorna os conteúdos mais relacionados.

Essa funcionalidade amplia a capacidade do SCRIPTO, permitindo descoberta de conhecimento e reutilização de informações técnicas.

In [29]:
def recomendar_documentos(indice_documento, top_n=5):

    # Embedding do documento selecionado
    embedding_documento = embeddings_base[indice_documento].reshape(1, -1)

    # Similaridade com todos os documentos
    similaridades = cosine_similarity(
        embedding_documento,
        embeddings_base
    )[0]

    # Ordenar do maior para o menor
    indices = np.argsort(similaridades)[::-1]

    resultados = []

    for indice in indices:

        # Não recomendar o próprio documento
        if indice == indice_documento:
            continue

        resultados.append({
            "projeto": metadados.iloc[indice]["repo_full_name"],
            "documento": metadados.iloc[indice]["file_name"],
            "categoria": metadados.iloc[indice]["categoria_documento"],
            "similaridade": round(float(similaridades[indice]) * 100, 2)
        })

        if len(resultados) == top_n:
            break

    return resultados

In [30]:
recomendar_documentos(0)

[{'projeto': 'crccheck/docker-hello-world',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 46.32},
 {'projeto': 'mingfang/docker-predictionio',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 42.12},
 {'projeto': 'svhawks/Sendy-Docker',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 41.44},
 {'projeto': 'eBayClassifiedsGroup/PanteraS',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 39.78},
 {'projeto': 'dzove855/Bash-web-server',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 35.67}]

# Validação do Documento Selecionado

Antes de gerar recomendações, será validado qual documento está associado ao índice utilizado.

Essa verificação garante que o embedding selecionado corresponde corretamente ao documento apresentado ao usuário.

In [31]:
metadados.iloc[0]

,0
repo_full_name,0xN1nja/homepagectl
file_name,README.md
categoria_documento,Documentacao
content,# homepagectl\n\nTired of manually writing hom...
indice_embedding,0


In [32]:
metadados.sort_values(
    "content",
    key=lambda x: x.str.len(),
    ascending=False
).head(1)[
    [
        "repo_full_name",
        "file_name",
        "categoria_documento",
        "indice_embedding"
    ]
]

,repo_full_name,file_name,categoria_documento,indice_embedding
836,sudheerj/reactjs-interview-questions,README.md,Documentacao,836


# Teste de Recomendação com Documento Técnico Extenso

Nesta etapa será validada a funcionalidade de recomendação utilizando um documento com maior volume de informações técnicas.

O objetivo é verificar se o modelo consegue identificar conteúdos relacionados a partir do contexto semântico de um documento completo.

Documento utilizado no teste:

- Repositório: sudheerj/reactjs-interview-questions
- Arquivo: README.md
- Índice do embedding: 836

A expectativa é que o modelo encontre documentos com temas próximos, como desenvolvimento web, programação, frameworks e tecnologias relacionadas.

In [33]:
recomendar_documentos(836)

[{'projeto': 'typescript-cheatsheets/react',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 53.38},
 {'projeto': 'chris-sev/make-react-apps',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 48.49},
 {'projeto': 'jestjs/jest',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 48.48},
 {'projeto': 'pmndrs/react-nil',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 47.78},
 {'projeto': 'hasan-py/Hayroo',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 47.59}]

# Filtro de Relevância das Recomendações

Nesta etapa será aplicado um filtro mínimo de similaridade para melhorar a qualidade das recomendações.

O objetivo é evitar que documentos pouco relacionados sejam apresentados ao usuário.

Critério utilizado:

- Documentos com similaridade abaixo de 35% serão descartados;
- Apenas conteúdos considerados semanticamente relacionados serão retornados.

Essa etapa melhora a experiência do usuário e aumenta a precisão da recomendação.

In [34]:
def recomendar_documentos(indice_documento, top_n=5, limite_similaridade=35):

    # Embedding do documento selecionado
    embedding_documento = embeddings_base[indice_documento].reshape(1, -1)

    # Calcula similaridade com todos os documentos
    similaridades = cosine_similarity(
        embedding_documento,
        embeddings_base
    )[0]

    # Ordena do maior para o menor
    indices = np.argsort(similaridades)[::-1]

    resultados = []

    for indice in indices:

        # Ignorar o próprio documento
        if indice == indice_documento:
            continue

        # Converter similaridade para percentual
        score = float(similaridades[indice]) * 100

        # Aplicar filtro de relevância
        if score < limite_similaridade:
            continue

        resultados.append({
            "projeto": metadados.iloc[indice]["repo_full_name"],
            "documento": metadados.iloc[indice]["file_name"],
            "categoria": metadados.iloc[indice]["categoria_documento"],
            "similaridade": round(score, 2)
        })

        if len(resultados) == top_n:
            break

    return resultados

In [35]:
recomendar_documentos(836)

[{'projeto': 'typescript-cheatsheets/react',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 53.38},
 {'projeto': 'chris-sev/make-react-apps',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 48.49},
 {'projeto': 'jestjs/jest',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 48.48},
 {'projeto': 'pmndrs/react-nil',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 47.78},
 {'projeto': 'hasan-py/Hayroo',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 47.59}]

# Apresentação dos Resultados - SCRIPTO

Nesta etapa será criada uma camada de apresentação dos resultados gerados pelo modelo de IA.

O objetivo é transformar a saída técnica da busca semântica em uma visualização mais amigável para usuários e para futura integração com a interface do sistema.

A lógica de recomendação permanece separada da camada de apresentação, permitindo que o Backend ou Frontend utilize os dados conforme a necessidade.

In [36]:
def exibir_recomendacoes(resultados):

    print("🔎 Conteúdos relacionados encontrados\n")

    print("=" * 50)

    for posicao, item in enumerate(resultados, start=1):

        if posicao == 1:
            medalha = "🥇"
        elif posicao == 2:
            medalha = "🥈"
        elif posicao == 3:
            medalha = "🥉"
        else:
            medalha = "📌"

        print(f"{medalha} {posicao}º Resultado")
        print(f"📂 Projeto: {item['projeto']}")
        print(f"📄 Documento: {item['documento']}")
        print(f"🏷 Categoria: {item['categoria']}")
        print(f"⭐ Similaridade: {item['similaridade']}%")
        print("-" * 50)

In [37]:
resultados = recomendar_documentos(836)

exibir_recomendacoes(resultados)

🔎 Conteúdos relacionados encontrados

🥇 1º Resultado
📂 Projeto: typescript-cheatsheets/react
📄 Documento: README.md
🏷 Categoria: Documentacao
⭐ Similaridade: 53.38%
--------------------------------------------------
🥈 2º Resultado
📂 Projeto: chris-sev/make-react-apps
📄 Documento: README.md
🏷 Categoria: Documentacao
⭐ Similaridade: 48.49%
--------------------------------------------------
🥉 3º Resultado
📂 Projeto: jestjs/jest
📄 Documento: README.md
🏷 Categoria: Documentacao
⭐ Similaridade: 48.48%
--------------------------------------------------
📌 4º Resultado
📂 Projeto: pmndrs/react-nil
📄 Documento: README.md
🏷 Categoria: Documentacao
⭐ Similaridade: 47.78%
--------------------------------------------------
📌 5º Resultado
📂 Projeto: hasan-py/Hayroo
📄 Documento: README.md
🏷 Categoria: Documentacao
⭐ Similaridade: 47.59%
--------------------------------------------------


# Classificação de Relevância dos Resultados

Nesta etapa será criada uma classificação interpretativa da similaridade encontrada pelo modelo.

O objetivo é transformar o valor matemático de similaridade em uma informação mais compreensível para o usuário final.

Critérios:

- Acima de 80%: Muito relacionada;
- Entre 60% e 79%: Relacionada;
- Entre 35% e 59%: Possível relação.

Essa camada facilita a interpretação dos resultados apresentados pelo SCRIPTO.

In [38]:
def classificar_relevancia(score):

    if score >= 80:
        return "Muito relacionada"

    elif score >= 60:
        return "Relacionada"

    else:
        return "Possível relação"


def exibir_recomendacoes(resultados):

    print("🔎 Conteúdos relacionados encontrados\n")

    print("=" * 50)

    for posicao, item in enumerate(resultados, start=1):

        if posicao == 1:
            medalha = "🥇"
        elif posicao == 2:
            medalha = "🥈"
        elif posicao == 3:
            medalha = "🥉"
        else:
            medalha = "📌"

        relevancia = classificar_relevancia(item["similaridade"])

        print(f"{medalha} {posicao}º Resultado")
        print(f"📂 Projeto: {item['projeto']}")
        print(f"📄 Documento: {item['documento']}")
        print(f"🏷 Categoria: {item['categoria']}")
        print(f"⭐ Similaridade: {item['similaridade']}%")
        print(f"📌 Relevância: {relevancia}")
        print("-" * 50)

In [39]:
resultados = recomendar_documentos(836)

exibir_recomendacoes(resultados)

🔎 Conteúdos relacionados encontrados

🥇 1º Resultado
📂 Projeto: typescript-cheatsheets/react
📄 Documento: README.md
🏷 Categoria: Documentacao
⭐ Similaridade: 53.38%
📌 Relevância: Possível relação
--------------------------------------------------
🥈 2º Resultado
📂 Projeto: chris-sev/make-react-apps
📄 Documento: README.md
🏷 Categoria: Documentacao
⭐ Similaridade: 48.49%
📌 Relevância: Possível relação
--------------------------------------------------
🥉 3º Resultado
📂 Projeto: jestjs/jest
📄 Documento: README.md
🏷 Categoria: Documentacao
⭐ Similaridade: 48.48%
📌 Relevância: Possível relação
--------------------------------------------------
📌 4º Resultado
📂 Projeto: pmndrs/react-nil
📄 Documento: README.md
🏷 Categoria: Documentacao
⭐ Similaridade: 47.78%
📌 Relevância: Possível relação
--------------------------------------------------
📌 5º Resultado
📂 Projeto: hasan-py/Hayroo
📄 Documento: README.md
🏷 Categoria: Documentacao
⭐ Similaridade: 47.59%
📌 Relevância: Possível relação
-------------

# Refinamento da Apresentação dos Resultados - SCRIPTO

Nesta etapa será aprimorada a apresentação das recomendações geradas pelo modelo de IA.

O objetivo é transformar os valores de similaridade em níveis de relevância mais intuitivos para o usuário final.

Também será apresentado o documento consultado antes da lista de recomendações, criando uma experiência mais próxima da utilização real da plataforma SCRIPTO.

In [41]:
def classificar_relevancia(score):

    if score >= 90:
        return "🟢 Altamente relacionada"

    elif score >= 70:
        return "🟡 Relacionada"

    elif score >= 50:
        return "🔵 Moderadamente relacionada"

    else:
        return "⚪ Baixa relação"

# Expansão da Base de Conhecimento - Todos os Documentos Técnicos

Inicialmente o modelo foi desenvolvido utilizando apenas arquivos README.md, devido à concentração de informações descritivas nesses documentos.

Nesta etapa, a base de conhecimento será expandida para incluir todos os documentos técnicos disponíveis no dataset tratado.

Serão considerados documentos como:

- README.md;
- LICENSE;
- CONTRIBUTING.md;
- CHANGELOG.md;
- SECURITY.md;
- arquivos de configuração de dependências e linguagens.

Objetivo:

Aumentar a capacidade do SCRIPTO de encontrar, classificar e recomendar conteúdos técnicos relacionados, permitindo uma visão mais ampla do conhecimento disponível nos repositórios analisados.

In [44]:
import pandas as pd

caminho_dataset = "/content/drive/MyDrive/Hackathon_ONE/dataset/tratado/documentation_tratado.parquet"

dataset = pd.read_parquet(caminho_dataset)

print("Dataset carregado!")
print(dataset.shape)

dataset.head()

Dataset carregado!
(2879, 10)


,id,repo_full_name,file_name,content,file_size,readme_quality_score,collected_at,num_caracteres,conteudo_repetido,categoria_documento
0,2454,0xN1nja/homepagectl,README.md,# homepagectl\n\nTired of manually writing hom...,2041,42,2026-06-27 19:14:23.602441,2041,False,Documentacao
1,2455,0xN1nja/homepagectl,go.mod,module github.com/0xN1nja/homepagectl\n\ngo 1....,242,0,2026-06-27 19:14:43.368223,242,False,Configuracao_Linguagem
2,2158,0xNyk/council-of-high-intelligence,CHANGELOG.md,# Changelog\n\nAll notable changes to this pro...,5454,0,2026-06-27 18:22:28.437941,5454,False,Historico_Alteracoes
3,2157,0xNyk/council-of-high-intelligence,CONTRIBUTING.md,# Contributing\n\nThanks for contributing to C...,941,0,2026-06-27 18:22:26.587716,941,False,Contribuicao
4,2156,0xNyk/council-of-high-intelligence,LICENSE,MIT License\n\nCopyright (c) 2026 nyk\n\nPermi...,1060,0,2026-06-27 18:22:24.846284,1060,False,Licenca


# Análise da Base Completa de Documentos Técnicos

Nesta etapa será analisada a distribuição dos documentos após a expansão da base de conhecimento.

A validação permite verificar quais tipos de conteúdos técnicos serão utilizados pelo modelo de IA.

A base contempla diferentes categorias, como documentação, licenças, dependências e arquivos de configuração.

In [47]:
dataset["categoria_documento"].value_counts()

,count
categoria_documento,
Documentacao,941
Licenca,645
Dependencias,322
Contribuicao,262
Historico_Alteracoes,195
Governanca,130
Seguranca,122
Configuracao_Linguagem,118
Configuracao_Python,72


# Geração de Embeddings da Base Completa - SCRIPTO

Nesta etapa será aplicado o modelo de linguagem Sentence Transformer em todos os documentos técnicos tratados.

Cada documento será convertido em um vetor numérico (embedding), permitindo que o sistema compare conteúdos por similaridade semântica.

Processo:

1. Leitura dos 2.879 documentos técnicos;
2. Conversão do conteúdo textual em embeddings;
3. Armazenamento dos vetores gerados;
4. Salvamento dos metadados para consulta futura.

Resultado esperado:

- 2.879 embeddings gerados;
- Vetores com 384 dimensões;
- Base preparada para busca semântica e recomendação automática.

In [48]:
from sentence_transformers import SentenceTransformer
import numpy as np

# Carregar modelo
modelo = SentenceTransformer(
    "sentence-transformers/all-MiniLM-L6-v2"
)

print("Modelo carregado!")

Loading weights:   0%|          | 0/103 [00:00<?, ?it/s]

Modelo carregado!


In [49]:
# Gerar embeddings de todos os documentos

textos = dataset["content"].tolist()

embeddings_documentos = modelo.encode(
    textos,
    batch_size=32,
    show_progress_bar=True
)

print("Embeddings gerados!")
print("Quantidade de documentos:", len(embeddings_documentos))
print("Dimensão dos vetores:", embeddings_documentos.shape[1])

Batches:   0%|          | 0/90 [00:00<?, ?it/s]

Embeddings gerados!
Quantidade de documentos: 2879
Dimensão dos vetores: 384


# Armazenamento dos Embeddings e Metadados - SCRIPTO

Após a geração dos embeddings, os vetores serão armazenados para evitar o processamento novamente a cada execução.

Também serão salvos os metadados dos documentos, permitindo relacionar cada vetor ao seu repositório, arquivo e categoria.

Arquivos gerados:

- embeddings_documentos_tecnicos.npy
  - Contém os vetores numéricos gerados pelo modelo de IA.

- documentos_tecnicos.csv
  - Contém as informações dos documentos para consulta e identificação dos resultados.

In [50]:
import os

pasta_embeddings = "/content/drive/MyDrive/Hackathon_ONE/SCRIPTO_AI/embeddings"
pasta_metadados = "/content/drive/MyDrive/Hackathon_ONE/SCRIPTO_AI/metadados"

os.makedirs(pasta_embeddings, exist_ok=True)
os.makedirs(pasta_metadados, exist_ok=True)

print("Estrutura SCRIPTO_AI criada!")

Estrutura SCRIPTO_AI criada!


In [51]:
caminho_embeddings = (
    "/content/drive/MyDrive/Hackathon_ONE/"
    "SCRIPTO_AI/embeddings/"
    "embeddings_documentos_tecnicos.npy"
)

np.save(
    caminho_embeddings,
    embeddings_documentos
)

print("Embeddings salvos com sucesso!")
print(caminho_embeddings)

Embeddings salvos com sucesso!
/content/drive/MyDrive/Hackathon_ONE/SCRIPTO_AI/embeddings/embeddings_documentos_tecnicos.npy


In [52]:
metadados_documentos = dataset[
    [
        "repo_full_name",
        "file_name",
        "categoria_documento",
        "num_caracteres"
    ]
].copy()


caminho_metadados = (
    "/content/drive/MyDrive/Hackathon_ONE/"
    "SCRIPTO_AI/metadados/"
    "documentos_tecnicos.csv"
)


metadados_documentos.to_csv(
    caminho_metadados,
    index=False
)

print("Metadados salvos com sucesso!")
print(metadados_documentos.shape)

Metadados salvos com sucesso!
(2879, 4)


In [53]:
import os

print(os.path.exists(caminho_embeddings))
print(os.path.exists(caminho_metadados))

True
True


# Carregamento da Base Completa de IA - SCRIPTO

Nesta etapa será carregada a nova base de conhecimento contendo todos os documentos técnicos processados.

A versão anterior utilizava apenas arquivos README.md.

Nesta versão, o SCRIPTO utiliza todos os documentos classificados:

- README.md;
- LICENSE;
- CONTRIBUTING.md;
- CHANGELOG.md;
- SECURITY.md;
- arquivos de dependências;
- arquivos de configuração.

Essa expansão aumenta a capacidade de busca e recomendação semântica do sistema.

In [54]:
import numpy as np
import pandas as pd

# Caminhos da nova base

caminho_embeddings = (
    "/content/drive/MyDrive/Hackathon_ONE/"
    "SCRIPTO_AI/embeddings/"
    "embeddings_documentos_tecnicos.npy"
)

caminho_metadados = (
    "/content/drive/MyDrive/Hackathon_ONE/"
    "SCRIPTO_AI/metadados/"
    "documentos_tecnicos.csv"
)


# Carregar embeddings
embeddings_base = np.load(caminho_embeddings)


# Carregar metadados
metadados = pd.read_csv(caminho_metadados)


print("Base completa carregada!")
print("Embeddings:", embeddings_base.shape)
print("Metadados:", metadados.shape)

Base completa carregada!
Embeddings: (2879, 384)
Metadados: (2879, 4)


# Busca Semântica na Base Completa - SCRIPTO

Nesta etapa será atualizada a função de busca inteligente utilizando todos os documentos técnicos disponíveis.

O usuário poderá realizar consultas em linguagem natural e o sistema irá comparar o significado da pesquisa com os embeddings de todos os documentos.

A busca considera diferentes tipos de arquivos técnicos, permitindo encontrar conteúdos relacionados mesmo quando os termos utilizados são diferentes.

In [55]:
from sklearn.metrics.pairwise import cosine_similarity


def buscar_scripto(pergunta, top_n=5):

    # Transformar pergunta em embedding
    embedding_pergunta = modelo.encode([pergunta])


    # Comparar com todos os documentos
    similaridades = cosine_similarity(
        embedding_pergunta,
        embeddings_base
    )[0]


    # Selecionar maiores similaridades
    indices = np.argsort(similaridades)[::-1][:top_n]


    resultados = []

    for indice in indices:

        resultados.append({
            "projeto": metadados.iloc[indice]["repo_full_name"],
            "documento": metadados.iloc[indice]["file_name"],
            "categoria": metadados.iloc[indice]["categoria_documento"],
            "similaridade": round(
                float(similaridades[indice]) * 100,
                2
            )
        })


    return resultados

In [56]:
buscar_scripto(
    "Como instalar dependências de um projeto Python"
)

[{'projeto': 'microsoft/qlib',
  'documento': 'pyproject.toml',
  'categoria': 'Configuracao_Python',
  'similaridade': 51.08},
 {'projeto': 'taehoonlee/tensornets',
  'documento': 'pyproject.toml',
  'categoria': 'Configuracao_Python',
  'similaridade': 50.77},
 {'projeto': 'pyenv/pyenv',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 49.91},
 {'projeto': 'faif/python-patterns',
  'documento': 'pyproject.toml',
  'categoria': 'Configuracao_Python',
  'similaridade': 49.67},
 {'projeto': 'heroku/heroku-buildpack-python',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 47.05}]

In [57]:
buscar_scripto(
    "Como contribuir com um projeto open source"
)

[{'projeto': 'birl-language/birl-server',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 43.66},
 {'projeto': 'Aliucord/Aliucord',
  'documento': 'LICENSE',
  'categoria': 'Licenca',
  'similaridade': 37.94},
 {'projeto': 'WaxMoon/MultiApp',
  'documento': 'LICENSE',
  'categoria': 'Licenca',
  'similaridade': 37.43},
 {'projeto': 'liquidprompt/liquidprompt',
  'documento': 'LICENSE',
  'categoria': 'Licenca',
  'similaridade': 37.43},
 {'projeto': 'triwinds/ns-emu-tools',
  'documento': 'LICENSE',
  'categoria': 'Licenca',
  'similaridade': 37.43}]

In [58]:
metadados["categoria_documento"].value_counts()

,count
categoria_documento,
Documentacao,941
Licenca,645
Dependencias,322
Contribuicao,262
Historico_Alteracoes,195
Governanca,130
Seguranca,122
Configuracao_Linguagem,118
Configuracao_Python,72


# Ranking Inteligente por Categoria - SCRIPTO

Nesta etapa será adicionada uma camada de inteligência ao ranking dos resultados.

Além da similaridade semântica calculada pelo modelo de IA, serão considerados os tipos de documentos encontrados.

O objetivo é aumentar a relevância dos resultados de acordo com a intenção da busca.

Exemplo:

Uma busca relacionada a contribuição em projetos open source deve priorizar:

- CONTRIBUTING.md;
- CODE_OF_CONDUCT.md;
- documentos de Governança.

Essa abordagem combina inteligência artificial com regras de negócio para melhorar a experiência do usuário.

In [59]:
def aplicar_peso_categoria(categoria, pergunta):

    pergunta = pergunta.lower()

    peso = 1.0


    # Busca relacionada a contribuição
    if any(
        palavra in pergunta
        for palavra in [
            "contribuir",
            "contribuição",
            "open source",
            "colaborar"
        ]
    ):

        if categoria == "Contribuicao":
            peso = 1.25

        elif categoria == "Governanca":
            peso = 1.15

        elif categoria == "Licenca":
            peso = 0.90


    # Busca relacionada a dependências
    elif any(
        palavra in pergunta
        for palavra in [
            "dependência",
            "instalar",
            "pacote",
            "biblioteca"
        ]
    ):

        if categoria == "Dependencias":
            peso = 1.25


    return peso

In [60]:
def buscar_scripto_inteligente(pergunta, top_n=5):

    embedding_pergunta = modelo.encode([pergunta])

    similaridades = cosine_similarity(
        embedding_pergunta,
        embeddings_base
    )[0]


    resultados = []


    for indice, similaridade in enumerate(similaridades):

        categoria = metadados.iloc[indice]["categoria_documento"]

        peso = aplicar_peso_categoria(
            categoria,
            pergunta
        )


        score_final = similaridade * peso * 100


        resultados.append({

            "projeto": metadados.iloc[indice]["repo_full_name"],
            "documento": metadados.iloc[indice]["file_name"],
            "categoria": categoria,
            "similaridade": round(float(score_final),2)

        })


    resultados = sorted(
        resultados,
        key=lambda x: x["similaridade"],
        reverse=True
    )


    return resultados[:top_n]

In [61]:
buscar_scripto_inteligente(
    "Como contribuir com um projeto open source"
)

[{'projeto': 'birl-language/birl-server',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 43.66},
 {'projeto': 'microsoft/Tutel',
  'documento': 'CODE_OF_CONDUCT.md',
  'categoria': 'Governanca',
  'similaridade': 37.63},
 {'projeto': 'microsoft/qlib',
  'documento': 'CODE_OF_CONDUCT.md',
  'categoria': 'Governanca',
  'similaridade': 37.63},
 {'projeto': 'HKUDS/nanobot',
  'documento': 'CONTRIBUTING.md',
  'categoria': 'Contribuicao',
  'similaridade': 37.27},
 {'projeto': 'openclaw/openclaw',
  'documento': 'CONTRIBUTING.md',
  'categoria': 'Contribuicao',
  'similaridade': 35.69}]

# Salvamento da Base Final de IA - SCRIPTO

Nesta etapa são armazenados os artefatos finais gerados pelo modelo de IA.

Esses arquivos serão utilizados posteriormente pelo Backend para integração com a aplicação.

Arquivos gerados:

- embeddings_documentos_tecnicos.npy:
  Contém os vetores semânticos dos documentos.

- documentos_tecnicos.csv:
  Contém os metadados utilizados para identificar os resultados.

Esses arquivos representam a base de conhecimento do SCRIPTO.

In [62]:
import os

pasta_final = "/content/drive/MyDrive/Hackathon_ONE/SCRIPTO_AI/modelo_final"

os.makedirs(
    pasta_final,
    exist_ok=True
)

print("Pasta final criada!")

Pasta final criada!


In [63]:
np.save(
    "/content/drive/MyDrive/Hackathon_ONE/SCRIPTO_AI/modelo_final/embeddings_scripto.npy",
    embeddings_base
)

print("Embeddings finais salvos!")

Embeddings finais salvos!


In [64]:
metadados.to_csv(
    "/content/drive/MyDrive/Hackathon_ONE/SCRIPTO_AI/modelo_final/metadados_scripto.csv",
    index=False
)

print("Metadados finais salvos!")

Metadados finais salvos!


# Validação Final do Modelo

Nesta etapa são realizados testes para verificar se o modelo consegue recuperar documentos relacionados a diferentes intenções de busca.

Os testes simulam possíveis consultas realizadas pelos usuários do SCRIPTO.

In [65]:
buscar_scripto_inteligente(
    "Como instalar dependências de um projeto Python"
)

[{'projeto': 'harlanhong/CVPR2022-DaGAN',
  'documento': 'requirements.txt',
  'categoria': 'Dependencias',
  'similaridade': 54.96},
 {'projeto': 'coqui-ai/TTS',
  'documento': 'requirements.txt',
  'categoria': 'Dependencias',
  'similaridade': 51.96},
 {'projeto': 'kel-z/HSR-Scanner',
  'documento': 'requirements.txt',
  'categoria': 'Dependencias',
  'similaridade': 51.55},
 {'projeto': 'microsoft/qlib',
  'documento': 'pyproject.toml',
  'categoria': 'Configuracao_Python',
  'similaridade': 51.08},
 {'projeto': 'taehoonlee/tensornets',
  'documento': 'pyproject.toml',
  'categoria': 'Configuracao_Python',
  'similaridade': 50.77}]

In [66]:
buscar_scripto_inteligente(
    "Como contribuir com um projeto open source"
)

[{'projeto': 'birl-language/birl-server',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 43.66},
 {'projeto': 'microsoft/Tutel',
  'documento': 'CODE_OF_CONDUCT.md',
  'categoria': 'Governanca',
  'similaridade': 37.63},
 {'projeto': 'microsoft/qlib',
  'documento': 'CODE_OF_CONDUCT.md',
  'categoria': 'Governanca',
  'similaridade': 37.63},
 {'projeto': 'HKUDS/nanobot',
  'documento': 'CONTRIBUTING.md',
  'categoria': 'Contribuicao',
  'similaridade': 37.27},
 {'projeto': 'openclaw/openclaw',
  'documento': 'CONTRIBUTING.md',
  'categoria': 'Contribuicao',
  'similaridade': 35.69}]

In [67]:
buscar_scripto_inteligente(
    "Como configurar uma aplicação Java"
)

[{'projeto': 'birl-language/birl-server',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 53.61},
 {'projeto': 'alibaba/p3c',
  'documento': 'README.md',
  'categoria': 'Documentacao',
  'similaridade': 44.11},
 {'projeto': 'JodaOrg/joda-time',
  'documento': 'pom.xml',
  'categoria': 'Configuracao_Java',
  'similaridade': 39.68},
 {'projeto': 'alibaba/easy-retry',
  'documento': 'pom.xml',
  'categoria': 'Configuracao_Java',
  'similaridade': 38.95},
 {'projeto': 'AthenZ/athenz',
  'documento': 'pom.xml',
  'categoria': 'Configuracao_Java',
  'similaridade': 38.34}]

# Resumo Técnico - Data/IA SCRIPTO

O módulo de Inteligência Artificial do SCRIPTO foi desenvolvido utilizando técnicas de Processamento de Linguagem Natural (PLN) e embeddings semânticos.

Processamento realizado:

- Limpeza e preparação de 2.879 documentos técnicos;
- Classificação automática por categoria;
- Geração de embeddings utilizando Sentence Transformer;
- Criação de mecanismo de busca semântica;
- Desenvolvimento de ranking inteligente considerando similaridade e categoria documental.

Resultado:

O modelo permite encontrar conteúdos técnicos relacionados considerando o significado dos textos, indo além de buscas tradicionais por palavras-chave.

Artefatos disponibilizados para integração:

- embeddings_scripto.npy
- metadados_scripto.csv

In [68]:
from google.colab import drive

drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [69]:
!find /content/drive/MyDrive -name "*Modelo*IA*.ipynb"

/content/drive/MyDrive/Hackathon_ONE/03_Modelo_IA_SCRIPTO.ipynb


In [70]:
!git clone https://github.com/No-Country-simulation/S1-BR-G9-TEAM07-TechMind.git

Cloning into 'S1-BR-G9-TEAM07-TechMind'...
remote: Enumerating objects: 91, done.
remote: Counting objects: 100% (91/91), done.
remote: Compressing objects: 100% (68/68), done.
remote: Total 91 (delta 14), reused 73 (delta 8), pack-reused 0 (from 0)
Receiving objects: 100% (91/91), 28.06 KiB | 1.56 MiB/s, done.
Resolving deltas: 100% (14/14), done.


In [72]:
!git branch -a

* main
  remotes/origin/HEAD -> origin/main
  remotes/origin/develop
  remotes/origin/develop-backend
  remotes/origin/develop-data
  remotes/origin/main


In [73]:
!git checkout develop-data

Branch 'develop-data' set up to track remote branch 'develop-data' from 'origin'.
Switched to a new branch 'develop-data'


In [74]:
!git branch

* develop-data
  main


In [75]:
!ls

README.md


In [76]:
import os

os.makedirs("data/ia", exist_ok=True)

print("Estrutura criada!")

Estrutura criada!


In [79]:
!ls data/ia

In [81]:
!ls -la

total 16
drwxr-xr-x 4 root root 4096 Jul 12 18:38 .
drwxr-xr-x 1 root root 4096 Jul 12 18:34 ..
drwxr-xr-x 3 root root 4096 Jul 12 18:38 data
drwxr-xr-x 8 root root 4096 Jul 12 18:36 .git
-rw-r--r-- 1 root root    0 Jul 12 18:36 README.md
